ArgusCore/src/main/java/com/salesforce/dva/argus/service/batch/CachedBasedBatchService.java

```
@@ -0,0 +1,157 @@
+        super(config);
+        requireArgument(cacheService != null, "Cache service cannot be null.");
+        requireArgument(metricQueueService != null, "Metric queue service cannot be null.");
+        _config = config;
+        _cacheService = cacheService;
+        _metricQueueService = metricQueueService;
+    }
+
+    //~ Methods **************************************************************************************************************************************
+
+    @Override
+    public BatchMetricQuery findBatchById(String id) {
+        String json = _cacheService.get(ROOT + id);
+        if (json == null) {
+            return null;
+        }
+        try {
+            Map<String, Object> batchData = MAPPER.readValue(json, Map.class);
+            BatchMetricQuery.Status status = BatchMetricQuery.Status.fromInt((Integer) batchData.get("status"));
+            BatchMetricQuery.Priority priority = BatchMetricQuery.Priority.fromInt((Integer) batchData.get("priority"));
+            int ttl = Integer.valueOf(batchData.get("ttl").toString());
+            long createdDate = Long.valueOf(batchData.get("createdDate").toString());
+            String ownerName = (String) batchData.get("ownerName");
+            String batchId = (String) batchData.get("batchId");
+            List<AsyncBatchedMetricQuery> queries = new ArrayList<>();
+
+            Long[] queueIds = MAPPER.readValue((String) batchData.get("queueIds"), Long[].class);
+            for (Long queueId: queueIds) {
+                queries.add(_metricQueueService.findQueryById(priority.toInt(), queueId));
+            }
+            BatchMetricQuery batch = new BatchMetricQuery(status, priority, ttl, createdDate, batchId, ownerName, queries);
+            return batch;
+        } catch (Exception ex) {
+            LOGGER.error("Exception in BatchMetricQuery construction from JSON: {}", ex.toString());
```

For the API/service interface methods, it's better to throw the exception rather than log and swallow.  Check out Marty Fowler's paper called Fail Fast if you haven't checked it out.